### PR TITLE
Added docs for swift language

### PR DESCRIPTION
**Pull request by Uzaaft**
Added docs for Swift language, and bump docs
Please provide comments for the docs created here.
Is the LSP setup too over-complicated? I took inspiration for the LSP setup from [abzcoding/lvim/ftplugin/python.lua](https://github.com/abzcoding/lvim/blob/main/ftplugin/python.lua)

### CODE REVIEW

**Comment by Uzaaft** [comment]
image

**Comment by Uzaaft** [comment]
Wow. I am an idiot.  xD Dumbing it down right now.

**Comment by Uzaaft** [comment]
New look: 
image

File: package.json [hunk 1]
```diff
@@ -21,8 +21,9 @@
   "author": "Christian Chiarulli",
   "license": "ISC",
   "dependencies": {
-    "@vuepress/plugin-search": "^2.0.0-beta.27",
-    "vuepress": "^2.0.0-beta.27",
-    "vuepress-plugin-mermaidjs": "^1.8.1"
+    "@vuepress/client": "^2.0.0-beta.35",
+    "@vuepress/plugin-search": "^2.0.0-beta.35",
+    "vuepress": "^2.0.0-beta.35",
+    "vuepress-plugin-mermaidjs": "^1.9.0"
```
Comment (kylo252) [inline package.json:27]: Can you move these changes to a separate PR? we can test them and revert them quickly then if there's a problem.

File: docs/languages/swift.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Configure Lunarvim
+
+The Lsp, and formatting does not work out of the box. To fix this, do the following:
+
+### Lsp Configuration
+Create a file called `~/.config/lvim/ftplugin/swift.lua`:
+
+```lua
+local opts = {}
```
Comment (kylo252) [inline docs/languages/swift.md:23]: Do swift projects usually include c/c++ files? what about a small mention of the `root_dir` pattern? 

https://github.com/neovim/nvim-lspconfig/blob/ec7119b166b16e681f663fcbf16b7139b38172ae/lua/lspconfig/server_configurations/sourcekit.lua#L4-L8

File: docs/languages/swift.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+## Install Language Server
+
+```vim
+:LspInstall sourcekit
+```
+
+
+## Configure Lunarvim
+
+The Lsp, and formatting does not work out of the box. To fix this, do the following:
```
Comment (kylo252) [inline docs/languages/swift.md:17]:
```suggestion
The `sourcekit` language server is disabled (overridden) by default because of the conflict with `clangd`
```

I also feel like we shouldn't let `clangd` run at the same time, but I'm not sure.

File: package.json (hunk 1 above)
Comment (Uzaaft) [reply package.json:27]: Sure thing. Although, lunarvim.org was not usable for me without the new packages added by me. Not sure why.

File: docs/languages/swift.md (hunk 3 above)
Comment (Uzaaft) [reply docs/languages/swift.md]: Running `lvim`(with my custom config forked from abz's) in the swift compiler project enables only these things for me: 
image

Clangd runs normally in other projects.

File: package.json (hunk 1 above)
Comment (Uzaaft) [reply package.json:27]: I use `pnpm`, so that might be the source of the issue, but I get this error when running ` lunarvim.org` out of the box.

File: docs/languages/swift.md (hunk 2 above)
Comment (Uzaaft) [reply docs/languages/swift.md:23]: I think `Sourcekit` can be used as a replacement to other C-based langs LSP's.
From the [SourceKit-LSP](https://github.com/apple/sourcekit-lsp): 
image
Adding mention about root_dir pattern.

Comment (Uzaaft) [reply docs/languages/swift.md:23]: Does commit 86ffafefbdcab5a1e7652596341c25dda42c4e92 fix this, or should I add some more info?

File: package.json (hunk 1 above)
Comment (Uzaaft) [reply package.json]: Reverted `package.json` in e6b8170a4ea5b8c4c550d713593077502291c1bc

File: docs/languages/swift.md (hunk 2 above)
Comment (kylo252) [reply docs/languages/swift.md:23]: @Uzaaft, it depends, I'm still wondering if clangd needs to be disabled when enabling this one. Please start the sourcekit server (probably open a swift file), then open a c/c++ file and see what happens and how it plays along with clangd.

Comment (kylo252) [reply docs/languages/swift.md:23]: It's mainly just to see if they both show up in `:LspInfo`.  Does go-to definition and other LSP features still work for these files? any apparent conflict errors happening? That sort of thing. If it still doesn't seem to be broken, then it probably isn't 😄

File: package.json (hunk 1 above)
Comment (kylo252) [reply package.json]: To clarify, I don't have a problem with updating package.json, in fact, we should probably do that anyway for security reasons. I just thought it might be better to do it separately.

Comment (Uzaaft) [reply package.json]: @kylo252 I understand that. Opening up a new PR for the bump and addition in package.json :)

File: docs/languages/swift.md (hunk 2 above)
Comment (Uzaaft) [reply docs/languages/swift.md:23]: They both show up, and no issues when I jumped from a swift file to a cpp file inside a hello world project and the swift compiler project

File: package.json (hunk 1 above)
Comment (Uzaaft) [reply package.json]: Turns out vuepress 2.0 requires some config of pnpm

**Review by kylo252 — APPROVED** [review]
@Uzaaft, thank you so much for all the effort! ❤️

**Comment by Uzaaft** [comment]
Thank you for all the help!
